tests/testthat/test-code_dependency.R

```diff
@@ -0,0 +1,174 @@
+
+  testthat::expect_identical(
+    get_code(q, deparse = FALSE, names = "b"),
+    c("a <- 1", "b <- a")
+  )
+})
+
+testthat::test_that("get_code extract code of a parent binding if used in a function", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- identity(x = a)")
+  q <- eval_code(q, "a <- 2")
+
+  testthat::expect_identical(
+    get_code(q, deparse = FALSE, names = "b"),
+    c("a <- 1", "b <- identity(x = a)")
+  )
+})
+
+testthat::test_that("get_code is possible to output the code for multiple objects", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- 2")
+  q <- eval_code(q, "c <- 3")
+
+  testthat::expect_identical(
+    get_code(q, deparse = FALSE, names = c("a", "b")),
+    c("a <- 1", "b <- 2")
+  )
+})
+
+testthat::test_that("get_code can't extract the code when using assign", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "assign('b', 5)")
+  q <- eval_code(q, "b <- b + 2")
+  testthat::expect_identical(
+    get_code(q, deparse = FALSE, names = "b"),
+    "b <- b + 2"
+  )
+})
+
+testthat::test_that("get_code can extract the code when using <<-", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- a")
+  q <- eval_code(q, "b <<- b + 2")
+  testthat::expect_identical(
+    get_code(q, deparse = FALSE, names = "b"),
+    c("a <- 1", "b <- a", "b <<- b + 2")
+  )
+})
+
+testthat::test_that("get_code extracts the code when using eval with object", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- 2")
+  q <- eval_code(q, "eval(expression({b <- b + 2}))")
+  testthat::expect_identical(
+    get_code(q, deparse = FALSE, names = "b"),
+    c("b <- 2", "eval(expression({\n    b <- b + 2\n}))")
+  )
+})
+
+
+# @effect ---------------------------------------------------------------------------------------------------------
+
+
+testthat::test_that("@effect cause to return this line for affected binding", {
+  q <- new_qenv()
+  q <- eval_code(
+    q,
+    "
+  a <- 1 # @effect b
+  b <- 2
+  "
+  )
+
+  testthat::expect_identical(
+    get_code(q, deparse = FALSE, names = "b"),
+    c("a <- 1", "b <- 2")
+  )
+})
+
+testthat::test_that(
+  "@effect does not return this line for affected binding
+  if object is not specificed in the same eval_code",
+  {
+    q <- new_qenv()
+    q <- eval_code(q, "a <- 1 # @effect b")
+    q <- eval_code(q, "b <- 2")
+
+    testthat::expect_identical(
+      get_code(q, deparse = FALSE, names = "b"),
+      c("b <- 2")
+    )
+  }
+)
+
+testthat::test_that(
+  "@effect returns this line for affected binding
+  if object is not specificed in the same eval_code
+  but it existed already in the qenv@env",
+  {
+    q <- new_qenv()
+    q <- eval_code(q, "a <- 1 ")
+    q <- eval_code(q, "b <- 2 # @effect a")
+
+    testthat::expect_identical(
+      get_code(q, deparse = FALSE, names = "a"),
+      c("a <- 1", "b <- 2")
+    )
+  }
+)
```

Comment:
It would be nice to add the test to make sure that lines affecting parent evaluated after cooccurrence are not included. Can we check the same with `@effect` instead of using `b <- a`? 
```suggestion
)


testthat::test_that(
  "lines affecting parent evaluated after co-occurrence are not included in get_code output",
  {
    q <- new_qenv()
    q <- eval_code(q, "a <- 1")
    q <- eval_code(q, "b <- a")
    q <- eval_code(q, "a <- 3")
    
    testthat::expect_identical(
      get_code(q, deparse = FALSE, names = "b"),
      c("a <- 1", "b <- a")
    )
  }
)
```

Reply:
https://github.com/insightsengineering/teal.code/pull/139/commits/ddfa2bfe72a6f1f433ed35e99ac53bc8c2634532